rpc/rpc_client.go

```diff
@@ -231,42 +283,50 @@ func Connect(connConfig Config, suppressRegister bool, dispatcherFuncs map[strin
 	return true
 }
 
-func reAttemptLogin(err error) bool {
-	if atomic.LoadUint32(&reLoginRunning) == 1 {
-		return false
-	}
-	atomic.StoreUint32(&reLoginRunning, 1)
-
-	rpcLoginMu.Lock()
-	if rpcLoadCount == 0 && !rpcEmergencyModeLoaded {
-		Log.Warning("[RPC Store] --> Detected cold start, attempting to load from cache")
-		Log.Warning("[RPC Store] ----> Found APIs... beginning emergency load")
-		rpcEmergencyModeLoaded = true
-		if emergencyModeLoadedCallback != nil {
-			go emergencyModeLoadedCallback()
-		}
-	}
-	rpcLoginMu.Unlock()
-
-	time.Sleep(time.Second * 3)
-	atomic.StoreUint32(&reLoginRunning, 0)
+// Login tries to login to the rpc sever. Returns true if it succeeds and false
+// if it fails.
+func Login() bool {
+	// I know this is extreme but rpc.Login() appears about 17 times and teh
+	// methods appears to be sometimes called in goroutines.
+	//
+	// Unless someone audits to ensure all of where this appears the parent calls
+	// are not concurrent, this is a much safer solution.
+	v, _, _ := loginFlight.Do("Login", func() (interface{}, error) {
+		return loginBase(), nil
+	})
+	return v.(bool)
+}
 
-	if strings.Contains(err.Error(), "Cannot obtain response during timeout") {
-		reConnect()
```

Comment:
This is a no-op. Which doesn't help  much here as we will never recover from the emergency mode when we get timeout from connecting to remote rpc server.